android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -95,21 +168,74 @@ public object WorkerBinder {
    *   [Worker.coroutineContext] is not overriden with a value different than
    *   [EmptyCoroutineContext]
    */
+  @Deprecated(
+    message =
+      """
+      This method will bind on current thread where WorkerBinder is called. This often means that
+      the binding will happen on the main Thread leading to potential Jank issues/ANRs
+    """,
+    replaceWith = ReplaceWith("bindToInteractor(interactor, workers)"),
+  )
   @JvmStatic
   @JvmOverloads
   public fun bind(
     interactor: Interactor<*, *>,
     workers: List<Worker>,
-    dispatcherAtBinder: CoroutineDispatcher = RibDispatchers.Unconfined,
+    originalDispatcher: CoroutineDispatcher = RibDispatchers.Unconfined,
   ) {
+    val dispatcherAtBinder = getDispatcherFromMigration(originalDispatcher)
     for (interactorWorker in workers) {
       bind(interactor, interactorWorker, dispatcherAtBinder)
     }
   }
 
   /**
-   * Bind a worker (ie. a manager or any other class that needs an presenter's lifecycle) to an
-   * presenter's lifecycle events. Inject this class into your presenter and call this method on any
+   * Bind a worker (ie. a manager or any other class that needs an presenter's lifecycle) to a
+   * presenter's lifecycle events.
+   *
+   * IMPORTANT: Binding will happen on [RibDispatchers.Default] unless the Worker explicitly
+   * overrides [coroutineContext] to other than [EmptyCoroutineContext]
+   *
+   * @param presenter The presenter that provides the lifecycle.
+   * @param worker The class that wants to be informed when to start and stop doing work.
+   * @return [WorkerUnbinder] to unbind [Worker]'s lifecycle.
+   */
+  @JvmStatic
+  public fun bindToPresenter(
+    presenter: Presenter,
+    worker: Worker,
+  ): WorkerUnbinder =
+    worker.bind(
+      presenter.lifecycleFlow,
+      Presenter.lifecycleRange,
+      RibDispatchers.Default,
+      workerBinderListenerWeakRef,
+    )
+
+  /**
+   * Bind a worker (ie. a manager or any other class that needs an presenter's lifecycle) to a
+   * presenter's lifecycle events.
+   *
+   * IMPORTANT: Binding will happen on [RibDispatchers.Default] unless the Worker explicitly
+   * overrides [coroutineContext] to other than [EmptyCoroutineContext]
+   *
+   * @param presenter The presenter that provides the lifecycle.
+   * @param workers A list of classes that want to be informed when to start and stop doing work.
+   * @return [WorkerUnbinder] to unbind [Worker]'s lifecycle.
+   */
+  @JvmStatic
+  public fun bindToPresenter(
```

Comment:
This looks like new functionality added that isn't needed for these changes (unless I missed it's usage) where we support binding to a list.

1. Should this be a separate PR if we want a new puiblic API.
2. Rather in this diff or on the next, probably rename to bindToPresenters (plural)

Reply:
We've been providing these APIs (_**WorkerBinder.bind(interactor, listOfWorkers)/ WorkerBinder.bind(presenter, listOfWorkers**_) for a while. 

Part of the goal of this PR is to deprecate existing WorkerBinder.bind methods given that the methods with 2 parameter call takes a default RibDispatchers.Unconfined which often times results in binding on main thread. 

Was thinking that adding new methods (bindToInteractor, bindToPresenter) that are opinionated on using to RibDispatchers.Default to prevent further additions of WorkerBinder.bind on main thread (we can easily rely on static analysis to force everyone to use the new API when new WorkerBinder additions). 

Also another important caveat, if there is still a need for a legacy worker to be bound on the main thread this can be achieved by overriding Worker's coroutineContext property (this will ignore the default Dispatchers specified on WorkerBinder call)